e2e_test/udf/sql_udf.slt

```diff
@@ -0,0 +1,105 @@
+statement ok
+SET RW_IMPLICIT_FLUSH TO true;
+
+# Create an anonymous function with double dollar as clause
+statement ok
+create function add(INT, INT) returns int as $$select $1 + $2$$ language sql;
+
+# Create an anonymous function with single quote as clause
+statement ok
+create function sub(INT, INT) returns int as 'select $1 - $2' language sql;
+
+# Currently we can only support constant calling convention
```

Comment:
What does "constant calling convention" mean? 

And please also add a test case for the thing we don't support now, since you mentioned "we can only" 😄

Reply:
The "constant calling convention" here means that currently, we could call other pre-defined sql udfs in a new sql udf definition, but only supports constant parameters rather than anonymous / named parameters.
e.g., 
- `create function sub_1_1() returns int as 'select sub(1, 1)' language sql;` is supported. (Assuming `sub(INT, INT)` is already defined)
-`create function sub_wrapper(INT, INT) returns int as 'select sub($1, $2)' language sql;` is NOT supported.

Reply:
And the reason of the error test case is that, under the current implementation, this kind of use will overflow the stack and quit RW. I'll try figuring out a graceful error handling solution / resolving the issue in future PRs.

Reply:
Wow, it seems like there is a flaw in the current code design. Does this mean that we should consider implementing some scope concepts?

Reply:
For recursive cases, since the semantics itself is not supported (we don't have the base case at the beginning), such definitions should be rejected during function creation time.
And for the above case, I'll try come up with a more graceful solution after this PR, maybe will involve some scope concepts 🤔

Reply:
Resolved now.